### ISSUE TITLE
Recover Kazam mux file to mp4

### ISSUE THREAD

**Issue by rizmut** [bookkojot/mp4fixer#6]
From [this](https://video.stackexchange.com/questions/22279/recover-broken-kazam-video-file-mux) question I see you to try your project. But actually I have zero undestanding about media format. How to do that exactly in command?

**Comment by bartwr** [comment]
If you did not finish your Kazam screen recording, you'll have an 0 bytes .movie file and a .movie.mux file with a bigger size. See screenshot below.

![image](https://user-images.githubusercontent.com/899234/47601473-7d658e80-d9d1-11e8-915f-264b5d279f5d.png)

To make use of [mp4fixer](https://github.com/bookkojot/mp4fixer#fix-your-mp4-file-for-free-with-perl-opensource-tool-wet-ultimate-solution) for recovering the video file from the .mux file, you need to do the following:

1. Run Kazam again and this time finish the video.

Now you have three files:

- badvideo.movie
- badvideo.movie.mux
- goodexamplevideo.movie

Now, by following the [README](https://github.com/bookkojot/mp4fixer#how-to-use-it) of this project, you can rebuild the movie file from the .movie.mux file:

`perl fixer.pl goodexamplevideo.movie badvideo.movie.mux recovered`

The new generated file `recovered-out-video.h264` will be the recovered video file. Getting sound is [explained](https://github.com/bookkojot/mp4fixer#how-it-works) in the README as well.

**Comment by kiemnx** [comment]
I followed your guidline but the recovered-out-video.h264 file is empty (size 0 byte).
Step 1: record new file with same condition and length larger 20 seconds.
Step 2: finish recording (but not save file), the current record is kazam_nlajzhon.movie
Step 3: Run command:
`perl fixer.pl kazam_nlajzhon badvideo.movie.mux recovered`
ouput console:
`testing at 1aedd668 (of 827b80) gives us 1228935225
testing at 1ae25c2a (of 827b80) gives us 3858650738
testing at 1ad69d4c (of 827b80) gives us 3489495011
....
`
And finally, output console display:
`File: recovered-out-video.h264
  Size: 0         	Blocks: 0          IO Block: 4096   regular empty file
Device: 812h/2066d	Inode: 1839183     Links: 1
`
Did I have any mistakes?

**Comment by lamidesbetes** [comment]
Don't forget to install ffmeg ;)

sudo apt install ffmpeg

(Moreover, using sudo for the perl command can make it works:VLC can't read it but youtube can upload it)

**Comment by rizmut** [comment]
Hello, I have encountered similar problem, but now I can not file the .mux file. The issue occured when I tried to save Kazam video but the Save button in file choose dialog did not work. I canceled the operation but Kazam window still opened with grayed out color. Then I close the dialog and got Kazam to the first state. I am not sure I have to report to Kazam dev or not but my question with regarding your amazing tools is there any chance I can recover my file?

This is my "saved" file. I can not open it with VLC

![gambar](https://user-images.githubusercontent.com/5697991/97294638-d72f1680-1880-11eb-8126-cbc939896c8b.png)

**Comment by paulomp90** [comment]
> Hello, I have encountered similar problem, but now I can not file the .mux file. The issue occured when I tried to save Kazam video but the Save button in file choose dialog did not work. I canceled the operation but Kazam window still opened with grayed out color. Then I close the dialog and got Kazam to the first state. I am not sure I have to report to Kazam dev or not but my question with regarding your amazing tools is there any chance I can recover my file?
> 
> This is my "saved" file. I can not open it with VLC
> 
> ![gambar](https://user-images.githubusercontent.com/5697991/97294638-d72f1680-1880-11eb-8126-cbc939896c8b.png)

We're you able to recover?
I had an error when i was recording, but I feel I will lose everything :/

**Comment by Setigi3** [comment]
Hello! 

I need some help. I followed the procedure step by step but the recovery.h264 file has 0 bytes which means I cannot convert it to mp4.

Can someone please suggest a solution?
![mp4fixer](https://user-images.githubusercontent.com/78880298/107571347-c9e03280-6bb8-11eb-8b6f-4941132ff34d.png)

**Comment by rizmut** [comment]
> > Hello, I have encountered similar problem, but now I can not file the .mux file. The issue occured when I tried to save Kazam video but the Save button in file choose dialog did not work. I canceled the operation but Kazam window still opened with grayed out color. Then I close the dialog and got Kazam to the first state. I am not sure I have to report to Kazam dev or not but my question with regarding your amazing tools is there any chance I can recover my file?
> > This is my "saved" file. I can not open it with VLC
> > ![gambar](https://user-images.githubusercontent.com/5697991/97294638-d72f1680-1880-11eb-8126-cbc939896c8b.png)
> 
> We're you able to recover?
> I had an error when i was recording, but I feel I will lose everything :/

Yes, just simply convert this ![gambar](https://user-images.githubusercontent.com/5697991/97294638-d72f1680-1880-11eb-8126-cbc939896c8b.png) file to MP4 with VLC, it solved my problem

**Comment by vmlnayak** [comment]
> Hello!
> 
> I need some help. I followed the procedure step by step but the recovery.h264 file has 0 bytes which means I cannot convert it to mp4.
> 
> Can someone please suggest a solution?
> ![mp4fixer](https://user-images.githubusercontent.com/78880298/107571347-c9e03280-6bb8-11eb-8b6f-4941132ff34d.png)

**Comment by Setigi3** [comment]
@vmlnayak , I wound up installing ffmpeg before running the perl command.
sudo apt install ffmpeg

Make sure you do that first.

**Comment by jguillem** [comment]
I have been able to retrieve the video from kazam, but it has no audio. What have you done to get it back?

**Comment by vmlnayak** [comment]
I missed that video

**Comment by Setigi3** [comment]
First I have used this website https://h264-to-mp4.file-converter-online.com/ to convert h264 to mp4 thn I have used vlc to merge the video and audio (recored raw audio).

You can do so following this video: https://www.youtube.com/watch?v=3h_Iq3a7ul8

**Comment by jguillem** [comment]
In my case I only hear noises with the raw audio file recovered.
I convert the h264 to mp4 with ffmpeg and I have the video without audio. The raw audio file (recovered-out-audio.raw) is inaudible (only noises).
I'm not able the raw audio file recovered to mp3 with the aacfixer executable or change the extension to mp3, and if I try to merge the raw with the mp4 with VLC, I only hear noises.

**Comment by Arof2** [comment]
Hi there. 
I followed the procedure but i encountered an Error saying: `Only ACC streams can ne muxed by the ADTS muxer
Could not write header for output file #0 (incorrect codec parameters ?): Invalid argument`
I'm clueless of what to do next as no one else seems to have encountered this error. For some reason only 4 files were created which do not include the audio or video file i need. These files are: `recovered-headers.acc`, `recovered-headers.h264`, `recovered-nals.txt`, `recovered-stat.mp4`.
My two original files are labeled: `kazam_0l245qzg.movie` `kazam_0l245qzg.movie.mux`
I dont have any experience with corrupted files so i dont know what the error message means. I hope someone can help me. Thx

**Comment by amzker** [comment]
guys, i don't even have mux files in my save location

**Comment by caiot5** [comment]
Yeah, it works, but the video gets artifacts where it was supposed to be static images, it shows where the image changes but the part of image which is static keeps frozen onscreen. So I guess that's something to do with mp4 compression, it should be fixable, but I don't know how.
